### PR TITLE
Hgi 237 update audience

### PR DESCRIPTION
**Pull request by Innovative-GauravKochar**
_A summary of your pull request, including the what change you're making and why._

## Testing

_Include any additional information about the testing you have completed to
ensure your changes behave as expected. For a speedy review, please check
any of the tasks you completed below during your testing._

- [ ] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [ ] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [ ] [Segmenters] Tested in the staging environment

### CODE REVIEW

File: packages/destination-actions/src/destinations/braze-cohorts/syncAudiences/index.ts [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+    },
+    device_id: {
+      label: 'Device ID',
+      description: 'The unique device Identifier',
+      type: 'string'
+    },
+    cohort_id: {
+      label: 'Cohort ID',
+      description: 'The Cohort Identifier',
+      type: 'string',
```
Comment (varadarajan-tw) [inline packages/destination-actions/src/destinations/braze-cohorts/syncAudiences/index.ts:48]: Shouldn't the type be 'hidden'? 
[Refer PRD Mappings table](https://paper.dropbox.com/doc/Braze-Cohorts-PRD--BuRL1u4T24ZHkb5iPuWfy~VxAg-3ovFfGfRWJIORAD31Z9Hc).

**Review by varadarajan-tw — COMMENTED** [review]
I have left few comments. Pls do reach out to me in case you need clarification regarding any of those.  I understand this is still in progress. So, happy to do another round of review after everything is complete.
Overall, the progress looks good!

File: packages/destination-actions/src/destinations/braze-cohorts/syncAudiences/index.ts (hunk 1 above)
Comment (Innovative-GauravKochar) [reply packages/destination-actions/src/destinations/braze-cohorts/syncAudiences/index.ts:48]: yes it is , made this for testing !
changed to hidden

File: packages/destination-actions/src/destinations/braze-cohorts/syncAudiences/index.ts [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+  // We should never hit this condition because at least an user_id or device_id
+  // or user_alias is required in each payload, but if we do, returning early
+  // rather than hitting Cohort's API (with no data) is more efficient.
+  // The monoservice will interpret this early return as a 200.
+
+  if (!users.hasAddUsers && !users.hasRemoveUsers) {
+    return
+  }
+
+  const res = await SyncAudiencesApiClient.batchUpdate(settings, payloads[0], users)
```
[the remark this answers is not in the document]
Comment (Innovative-GauravKochar) [reply packages/destination-actions/src/destinations/braze-cohorts/syncAudiences/index.ts:120]: Noted !Here we just need cohort_id , we can pass

File: packages/destination-actions/src/destinations/braze-cohorts/syncAudiences/index.ts [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+
+  // When User gets added or removed successfully , it will give 201 statua code other than there could be an error of invalid keys or some internal error.
+  if (res.status !== 201) {
+    throw new RetryableError('Error while attempting to sync audience to braze. This batch will be retried.')
```
[the remark this answers is not in the document]
Comment (Innovative-GauravKochar) [reply packages/destination-actions/src/destinations/braze-cohorts/syncAudiences/index.ts:124]: So in that case , we don't need this , Will remove this!

**Pull request merged**